src/models.py

```diff
@@ -0,0 +1,439 @@
+        self.source = source
+        self.type = mtype
+        self.version = version
+
+    def load_description(self):
+        self.description = self.get_description().list()
+
+    def get_description(self):
+        options = dict(uri=self.measure_uri)
+        q = '''SELECT locale_code, locale_name, description
+               FROM measure_description INNER JOIN locale USING(locale_code)
+               WHERE measure_uri = $uri
+               ORDER BY locale_code;'''
+        description = db.query(q, options)
+        return description if description else []
+
+    @staticmethod
+    def get_all():
+        try:
+            return db.select('measure')
+        except (Exception, psycopg2.DatabaseError) as error:
+            logger.error(error)
+            raise Error(FATAL)
+
+
+class Country(object):
+    def __init__(self, country_uri, country_code, country_name, continent):
+        self.country_uri = country_uri
+        self.country_code = country_code
+        self.country_name = country_name
+        self.continent_code = continent
+
+
+def result_to_event(r):
+    return Event(r["event_id"], r["work_uri"], r["measure_uri"],
+                 r["timestamp"], r["value"], r["event_uri"],
+                 r["country_uri"], r["uploader_uri"])
+
+
+def result_to_country(r):
+    return Country(r["country_uri"], r["country_code"], r["country_name"],
+                   r["continent_code"])
+
+
+def result_to_measure(r, description=False):
+    measure = Measure(r["measure_uri"], r["namespace"], r["source"], r["type"],
+                      r["version"])
+    if description:
+        measure.load_description()
+    return measure
+
+
+def results_to_events(results):
+    return [(result_to_event(e).__dict__) for e in results]
+
+
+def results_to_countries(results):
+    return [(result_to_country(e).__dict__) for e in results]
+
+
+def results_to_measures(results, description=False):
+    return [(result_to_measure(e, description).__dict__) for e in results]
+
+
+def results_to_measure_aggregation(results):
+    data = []
+    for r in results:
+        measure = Measure(r["measure_uri"], r["namespace"], r["source"],
+                          r["type"], r["version"])
+        measure.value = r["value"]
+        data.append(measure.__dict__)
+    return data
+
+
+def results_to_measure_country_aggregation(results):
+    data = []
+    countries = []  # temporary list of countries
+    last = len(results) - 1
+    i = 0
+    for r in results:
+        if i == 0:
+            # we can't do cur=results[0] outsise--it moves IterBetter's pointer
+            cur = r
+        if r["measure_uri"] != cur["measure_uri"]:
+            measure = result_to_measure(cur)
+            measure.data = countries
+            data.append(measure.__dict__)
+            countries = []
+            cur = r
+        country = result_to_country(r)
+        country.value = r["value"]
+        countries.append(country.__dict__)
+        if i == last:
```

Comment:
Minor: My eyes might be tired, but can't the code in this if-block just be run after the for-loop? 
I'm not sure I understand why we need to check `i==last`?

Reply:
Yes, this could be done outside the loop, but also inside - and I'd rather keep the logic inside it.